ArjunaJTS/integration/src/test/java/com/arjuna/ats/jta/distributed/server/impl/ServerImpl.java

```diff
@@ -229,31 +228,28 @@ public TransactionManager getTransactionManager() {
 	public Xid locateOrImportTransactionThenResumeIt(int remainingTimeout, Xid toResume) throws XAException, IllegalStateException, SystemException,
 			IOException {
 		Xid toReturn = null;
-		Transaction transaction = rootTransactionsAsSubordinate.get(new SubordinateXidImple(toResume));
+		Transaction transaction = null;
+		JBossXATerminator xaTerminator = transactionManagerService.getJbossXATerminator();
+
+		if (!ExtendedJBossXATerminator.class.isInstance(xaTerminator)) {
+			System.out.printf("ExtendedJBossXATerminator: FAIL not an instance");
+		} else {
+			ExtendedJBossXATerminator extendedJBossXATerminator = (ExtendedJBossXATerminator) xaTerminator;
+			transaction = extendedJBossXATerminator.getTransaction(toResume);
+			if (transaction == null)
+				System.out.printf("ExtendedJBossXATerminator: FAIL missing tx");
+		}
 		if (transaction == null) {
 			transaction = SubordinationManager.getTransactionImporter().getImportedTransaction(toResume);
```

Comment:
If we made the changes we discuss above to get getTransaction check subordinate manager this step can be removed from the transport which would be nice